CMakeLists.txt

```diff
@@ -163,8 +163,8 @@ endif()
 # tests
 #
 
-include(CTest)
 if(BUILD_TESTING)
+  include(CTest)
```

Comment:
@Mizux, currently it defaults to `ON` if the user doesn't specify `BUILD_TESTTING=ON/OFF`
https://github.com/Kitware/CMake/blob/master/Modules/CTest.cmake#L50
Is this by design or should we apply these changes?

Reply:
yup is by design, here we would like to follow the idiomatic CMake way i.e. use `include(CTest)` which provide/advertise the option `BUILD_TESTING` **and** also call `enable_testing()`.
So this call must be placed **before** the if statement.

Concerning the default to `ON`, our point is we follow the CTest default behaviour so if CTest set it ON it will be ON and can be changed if the CTest implementation change one day... again no surprise here for a distro maintainer accustomed to the CMake build system default behaviour...

ref: https://cmake.org/cmake/help/latest/module/CTest.html#module:CTest
ref: https://cmake.org/cmake/help/book/mastering-cmake/cmake/Help/guide/tutorial/index.html#adding-support-for-a-dashboard-step-8